src/twisted/names/common.py

```diff
@@ -78,79 +84,160 @@ def query(self, query, timeout=None):
             return defer.maybeDeferred(method, query.name.name, timeout)
 
     def _lookup(self, name, cls, type, timeout):
+        """
+        Umbrella defer-as-failed logic.
+        """
         return defer.fail(NotImplementedError("ResolverBase._lookup"))
 
     def lookupAddress(self, name, timeout=None):
+        """
+        Lookup up IPV4 address (A record request).
```

Comment:
I like the phrasing of these docstrings a bit better than the base ones in `IResolver`. Could you move these changes there?